mongodbforms/documents.py

```diff
@@ -116,7 +116,21 @@ def construct_instance(form, instance, fields=None, exclude=None):
         elif isinstance(f, ListField):
             list_field = getattr(instance, f.name)
             uploads = cleaned_data[f.name]
+            idx_to_pop = []
```

Comment:
Docstring/comments explaining what we're trying to achieve and how? As discussed the `enumerate` is a very important component and is slightly deviously hidden.